toplevel/genprintval.mli

```diff
@@ -71,3 +71,9 @@ module type S =
 
 module Make(O : OBJ)(_ : EVALPATH with type valu = O.t) :
          (S with type t = O.t)
+
+(* User-provided printers *)
+
+val printertypes_sig : Env.t -> Types.signature
+
+val printer_types : Types.signature -> Path.t * Path.t
```

Comment:
I would expose only a single function with signature `Env.t -> Env.t * Path.t * Path.t`; the intermediate signature is an implementation detail.